scripts/post-processing/1-run_normalization.R

```diff
@@ -114,70 +131,131 @@ genes <- dataset[, gene.col]
+  message(paste("Warning: there are", num.nas, 
+                "NA's in your dataset. Turning them into zeroes."))
+  dataset <- dplyr::mutate_all(dataset, zoo::na.aggregate)
+}
+
 #-----------------------Run each algorithm in the list-------------------------#
 # For each algorithm, run through this loop
 for (algorithm in opt$algorithm) {
+  
+  # Print out progress message:
+  cat("\n \n Running:", algorithm, "...")
+  
+  # Run normalization algorithms
+  if (algorithm == "scran") {
     
-    # Print out progress message:
-    cat("\n \n Running:", algorithm, "...")
+    # scater wants the data to be rounded
+    sce <- SingleCellExperiment::SingleCellExperiment(list(counts = round(as.matrix(dataset))))
     
-    # Run normalization algorithms
-    if (algorithm  == "scale"){
-        data.out <- as.data.frame(scale(dataset))
-        title <- "Scaled Expression"
-        
-    } else  if (algorithm == "log") {
-        data.out <- sign(dataset) * log2(1 + abs(dataset))
-        title <- "Log2 Expression"
-        
-    } else  if (algorithm == "voom") {
-        data.out <- as.data.frame(limma::voom(counts = dataset,
-                                              normalize.method = "quantile",
-                                              plot = FALSE)$E)
-        title <- "Log2 Expression (Voom)"
-        
-    } else if (algorithm == "tmm") {
-        data.dge <- edgeR::DGEList(counts = dataset)
-        data.dge <- edgeR::calcNormFactors(data.dge)
-        data.out <- as.data.frame(edgeR::cpm(data.dge, normalized.lib.sizes = TRUE,
-                                             log = TRUE))
-        title <- "Log2 Expression (TMM / edgeR)"
-        
-    } else if (algorithm == "deseq") {
-        data.colData <- data.frame(row.names = colnames(dataset))
-        data.dds <- DESeq2::DESeqDataSetFromMatrix(dataset, colData = data.colData,
-                                                   design = ~1)
-        data.dds <- DESeq2::estimateSizeFactors(data.dds)
-        data.out <- as.data.frame(DESeq2::counts(data.dds, normalized = TRUE))
-        data.out <- sign(data.out) * log2(1 + abs(data.out))
-        title <- "Log2 Expression (DEseq2)"
-        
-    } else if (algorithm == "vsd") {
-        data.colData <- data.frame(row.names = colnames(dataset))
-        data.dds <- DESeq2::DESeqDataSetFromMatrix(dataset, colData = data.colData,
-                                                   design = ~1)
-        data.out <- SummarizedExperiment::assay(
-                    DESeq2::varianceStabilizingTransformation(data.dds, 
-                                                              blind = TRUE))
-        title <- "Log2 Expression (varianceStabilizingTransformation - DEseq2)"
+    # Make some clusters
+    sce <- suppressWarnings(scran::computeSumFactors(sce))
+    neg.fact <- which(sce@int_colData@listData$size_factor < 0)
+    
+    # Dealing with negative size factors:
+    if (length(neg.fact) < 100) {
```

Comment:
I don't know that this is worth changing here, but design-wise I think it would be preferable to set some sensible default for this threshold (perhaps even calculating it based on the number of samples in the dataset) that can be overridden by a command line option.

Reply:
Yeah. You're right, this is not ideal. Changing how this is handled.